Sources/Clocks/UnimplementedClock.swift

```diff
@@ -101,17 +101,17 @@
     }
 
     public var now: Instant {
-      XCTFail("Unimplemented: \(self.name).now")
+      reportIssue("Unimplemented: \(self.name).now")
```

Comment:
Now we could pass the proper file-line source context here, but:

1. We need to pass them into the initializer.
2. We need to update the `.unimplemented` static to be a function. While this is technically a breaking change, `.unimplemented` only recently started working in Swift 6 beta since `any Clock` did not support it, so it might be an OK breaking change to make.